Userland/Libraries/LibWeb/Page/EventHandler.cpp

```diff
@@ -133,18 +134,50 @@ bool EventHandler::handle_mousewheel(Gfx::IntPoint const& position, unsigned int
     if (modifiers & KeyModifier::Mod_Shift)
         swap(wheel_delta_x, wheel_delta_y);
 
-    // FIXME: Support wheel events in nested browsing contexts.
+    bool handled_event = false;
 
-    auto result = paint_root()->hit_test(position.to_type<float>(), Painting::HitTestType::Exact);
-    if (result.has_value() && result->paintable->handle_mousewheel({}, position, buttons, modifiers, wheel_delta_x, wheel_delta_y))
-        return true;
+    RefPtr<Painting::Paintable> paintable;
+    if (m_mouse_event_tracking_layout_node) {
+        paintable = m_mouse_event_tracking_layout_node->paintable();
+    } else {
+        if (auto result = paint_root()->hit_test(position.to_type<float>(), Painting::HitTestType::Exact); result.has_value())
+            paintable = result->paintable;
+    }
 
-    if (auto* page = m_browsing_context.page()) {
-        page->client().page_did_request_scroll(wheel_delta_x * 20, wheel_delta_y * 20);
-        return true;
+    if (paintable) {
+        paintable->handle_mousewheel({}, position, buttons, modifiers, wheel_delta_x, wheel_delta_y);
+
+        JS::GCPtr<DOM::Node> node = paintable->mouse_event_target();
+        if (!node)
+            node = paintable->dom_node();
+
+        if (node) {
+            // FIXME: Support wheel events in nested browsing contexts.
+            if (is<HTML::HTMLIFrameElement>(*node)) {
+                return false;
+            }
+
+            auto* layout_node = &paintable->layout_node();
+            while (layout_node && node && !node->is_element() && layout_node->parent()) {
+                layout_node = layout_node->parent();
+                node = layout_node->dom_node();
+            }
+            if (!node || !layout_node) {
+                return false;
+            }
+
+            auto offset = compute_mouse_event_offset(position, *layout_node);
+            if (node->dispatch_event(*UIEvents::WheelEvent::create_from_platform_event(node->realm(), UIEvents::EventNames::wheel, offset.x(), offset.y(), position.x(), position.y(), wheel_delta_x, wheel_delta_y))) {
```

Comment:
We do have `buttons` available here so I think those should be passed in as part of the init. I wonder if the create function would look better if you created the init here and passed it as a struct instead of ~8 arguments but 🤷‍♂️

Reply:
I haven't added `buttons` as argument for `create_from_platform_event` because `MouseEvent` constructor expects only the clicked button as an input. But I just checked and it turned out that there are both `button` and `buttons` in `MouseEvent` according to the spec https://www.w3.org/TR/uievents/#idl-mouseevent.

Reply:
Ok, I added another commit where I added buttons property in MouseEvent